pages/code-review.md

```diff
@@ -114,12 +149,26 @@ When reviewing code, you should be able to check off each of the following:
     - [ ] Do the changes address the project's needs?
     - [ ] Do the changes respect the project's existing style?
     - [ ] Does the new code avoid reproducing existing functionality?
-    - [ ] Are functions as simple as possible?
+    - [ ] Are functions/classes as simple as possible?
     - [ ] Is the code as efficient as possible?
-    - [ ] Is the usage of each function clear?
+    - [ ] Is the usage of each function/class clear?
     - [ ] Have edge cases been considered and tested for?
-
-Use this bookmarklet code to add this checklist to your Pull Request comments: [Code Review Checklist](https://gist.github.com/cfarm/a4174fe6f775353a3115)
+    - [ ] Does the code represent a logical unit of work?
+    - [ ] Are there any glaring syntax errors that were missed?
+    - [ ] Are language constructs being utilized properly?
+    - [ ] Are there no unecessary premature optimizations?
+    - [ ] Are any frameworks/libraries being used leveraged properly?
+    - [ ] Are the changes clean and would they pass through a linter?
+    - [ ] Are there useful comments/documentation where needed?
+    - [ ] Is the commit message well-formed with succinct details about
+          what the change is intended for?
+
+In adition to this checklist, you should also pull down the actual changes
+and run them manually to put them through their paces.  Does the code
```

Comment:
I disagree that automated tests alone will catch any problems with functionality.  What if the code provides many good tests that all pass, yet still ultimately ends up being incorrect?  What if the functionality itself misses the mark despite being well-architected and working correctly?

The Dolores Landingham bot is a great recent example of this, where the functionality of selecting a time for messages to be sent is straightforward in theory but difficult to implement correctly.  Recall how many iterations it took of running the code through its paces to get it right regarding the time selection and time zone checking, and it's still not quite correct despite numerous tests accounting for a variety of edge cases! :-)  The day check we had to add on account of using UTC in particular, that is not something I would have accounted for at all had the code not been run outside of initial development - it just didn't even enter my purview while working on it initially.

I trust that the developer has run their code as well, but there will be some level of confirmation bias to contend with.  While the tests might cover a wide variety of use/edge cases they are no guarantee that the functionality itself is correct.  What if the tests are testing the wrong thing(s)?  What if the functionality developed is technically sound and correct implementation-wise, but misses the mark in delivering the correct end result or introduces some unintended side effect that the developer isn't aware of (perhaps some other sub-system is touched without their knowledge)?

Running the code with a fresh set of eyes is immensely helpful in uncovering these these harder-to-find issues and verifying that the code does work as expected.  I also think this exercise is helpful to the code reviewer in understanding what the code is doing and provides a broader context around how the code and and project are evolving, thus contributing to overall depth of knowledge.

I'm definitely curious why you're not sure about this recommendation, what are your thoughts?